### ISSUE TITLE
Windowed Materialized View using hop does not create overlapping windows.

### ISSUE THREAD

**Issue by RossRothenstine** [ClickHouse/ClickHouse#34044]
Hello.

Given the following schema, I would expect a Materialized View with hop defined to "hop" the "window interval" every "hop interval". Presently it seems to just hop the window interval instead of hopping the hop interval. 

You will see from my data below that from my windowed view, I am seeing only window_interval aligned results in the destination, and no results of hopping. It looks suspiciously like a tumble instead of a hop. 

Given the time period of 00:16 to 00:18, with a window interval of 60 seconds, and a hop interval of 10s, am I correct to assume the proper entries would contain records for [00:16:00, 00:17:00], [00:16:10, 00:17:10], [00:16:20, 00:17:20], etc.

```sql
:) SELECT version()

SELECT version()

Query id: f474ec7a-af51-4980-b5b9-55cd68c05224

┌─version()─┐
│ 22.1.3.7  │
└───────────┘

1 rows in set. Elapsed: 0.004 sec.

CREATE TABLE hop_data(
    timestamp Timestamp,
    val int
) Engine = TinyLog

CREATE TABLE hop_test(
    hop_start DateTime,
    hop_end DateTime,
    sum_val int
) Engine = TinyLog

CREATE WINDOW VIEW hop_test_wv
TO hop_test
AS
SELECT 
    hopStart(w_id) AS hop_start,
    hopEnd(w_id) AS hop_end,
    SUM(val) AS sum_val
FROM
    hop_data
GROUP BY hop(timestamp, INTERVAL 10 SECOND, INTERVAL 60 SECOND) AS w_id
SETTINGS allow_experimental_window_view = 1

-- Insert a few data points.
 INSERT INTO hop_data VALUES (now(), 1);
 
 :) SELECT * FROM hop_data;

SELECT *
FROM hop_data

Query id: e03f25fb-9b78-4f39-9cc2-81894ec74f7d

┌───────────timestamp─┬─val─┐
│ 2022-01-27 00:16:16 │   1 │
│ 2022-01-27 00:16:21 │   1 │
│ 2022-01-27 00:16:22 │   1 │
│ 2022-01-27 00:16:25 │   1 │
│ 2022-01-27 00:16:26 │   1 │
│ 2022-01-27 00:16:28 │   1 │
│ 2022-01-27 00:16:30 │   1 │
│ 2022-01-27 00:16:31 │   1 │
│ 2022-01-27 00:16:31 │   1 │
│ 2022-01-27 00:17:11 │   1 │
│ 2022-01-27 00:17:11 │   1 │
└─────────────────────┴─────┘

11 rows in set. Elapsed: 0.006 sec.

:) SELECT * FROM hop_test;

SELECT *
FROM hop_test

Query id: e7d8c3a8-c467-419e-8d01-3706dff1d2e1

┌───────────hop_start─┬─────────────hop_end─┬─sum_val─┐
│ 2022-01-27 00:16:00 │ 2022-01-27 00:17:00 │       9 │
│ 2022-01-27 00:17:00 │ 2022-01-27 00:18:00 │       2 │
└─────────────────────┴─────────────────────┴─────────┘
```

**Comment by ghughes** [comment]
It works if you use `WATERMARK=ASCENDING` and set `ALLOWED_LATENESS` to the window interval:

```sql
CREATE WINDOW VIEW hop_test_wv
TO hop_test
WATERMARK=ASCENDING ALLOWED_LATENESS=INTERVAL 60 SECOND
AS SELECT 
    hopStart(w_id) AS hop_start,
    hopEnd(w_id) AS hop_end,
    SUM(val) AS sum_val
FROM
    hop_data
GROUP BY hop(timestamp, INTERVAL 10 SECOND, INTERVAL 60 SECOND) AS w_id
SETTINGS allow_experimental_window_view = 1
```

**Comment by ghughes** [comment]
`STRICTLY_ASCENDING` (and `ALLOWED_LATENESS` set to the window interval) seems to work much better - I'm no longer seeing threads get stuck in `threadFuncFireProc`

@Vxider @ucasfl would you mind confirming whether this is the intended configuration for hopping windows?

**Comment by ghughes** [comment]
`STRICTLY_ASCENDING` (and `ALLOWED_LATENESS` set to the window interval) seems to work much better - I'm no longer seeing threads get stuck in `threadFuncFireProc`

@Vxider @ucasfl would you mind confirming whether this is the intended configuration for overlapping windows?

**Comment by Vxider** [comment]
It's cause by adding a wrong interval, I'll fix this ASAP

**Comment by den-crane** [comment]
fixed  #36861
